File: backend/app/schemas/stats.py
from pydantic import BaseModel


class OverviewStats(BaseModel):
    total_books: int
    total_pages: int
    completed_books: int
    reading_books: int
    pending_books: int
    avg_pages_per_book: float
    avg_reading_days: float


class FavoriteGenre(BaseModel):
    genre: str
    count: int


class TopAuthor(BaseModel):
    author: str
    count: int

File: frontend/src/types/stats.ts
export interface OverviewStats {
  total_books: number
  total_pages: number
  completed_books: number
  reading_books: number
  pending_books: number
  avg_pages_per_book: number
  avg_reading_days: number
}

export interface FavoriteGenre {
  genre: string
  count: number
}

export interface TopAuthor {
  author: string
  count: number
}
